WebPortal/Pages/Project/DataProject/ProjectTools/ProjectDatabase.razor

```diff
@@ -278,8 +278,11 @@ else
         if (ResourceID.HasValue)
         {
             _resource = await projectDbContext.Project_Resources2.FirstOrDefaultAsync(r => r.ResourceId == ResourceID.Value);
-            _databaseResource = _resource?.GetResourceObject<ProjectResource_Database>();
-            _existingRequestType = _resource?.ResourceType;
+            if (_resource.IsValid<ProjectResource_Database>())
+            {
+                _databaseResource = _resource?.GetResourceObject<ProjectResource_Database>();
+                _existingRequestType = _resource?.ResourceType;
```

Comment:
The _existingRequestType should be outside the condition

Reply:
fixed